test-go fails on `ubuntu:22.04`: undefined: syscall.SYS_FCNTL

relates to failure on ubuntu 22.04 against mips architecture: https://github.com/tonistiigi/xx/actions/runs/12201462889/job/34039923103#step:5:1063

```
#88 141.3 not ok 35 mipsle-hellogo
#88 141.3 # (from function `assert_success' in file bats-assert/src/assert.bash, line 114,
#88 141.3 #  from function `testHelloGO' in file test-go.bats, line 240,
#88 141.3 #  in test file test-go.bats, line 311)
#88 141.3 #   `testHelloGO' failed with status 2
#88 141.3 # 
#88 141.3 # -- command failed --
#88 141.3 # status : 2
#88 141.3 # output (2 lines):
#88 141.3 #   # runtime
#88 141.3 #   /usr/lib/go-1.18/src/runtime/os_linux.go:206:40: undefined: syscall.SYS_FCNTL
#88 141.3 # --
#88 141.3 # 
#88 141.9 not ok 36 mipsle-softfloat-hellogo
#88 141.9 # (from function `assert_success' in file bats-assert/src/assert.bash, line 114,
#88 141.9 #  from function `testHelloGO' in file test-go.bats, line 240,
#88 141.9 #  in test file test-go.bats, line 317)
#88 141.9 #   `testHelloGO' failed with status 2
#88 141.9 # 
#88 141.9 # -- command failed --
#88 141.9 # status : 2
#88 141.9 # output (2 lines):
#88 141.9 #   # runtime
#88 141.9 #   /usr/lib/go-1.18/src/runtime/os_linux.go:206:40: undefined: syscall.SYS_FCNTL
#88 141.9 # --
#88 141.9 # 
#88 142.4 not ok 37 mips64le-hellogo
#88 142.4 # (from function `assert_success' in file bats-assert/src/assert.bash, line 114,
#88 142.4 #  from function `testHelloGO' in file test-go.bats, line 240,
#88 142.4 #  in test file test-go.bats, line 323)
#88 142.4 #   `testHelloGO' failed with status 2
#88 142.4 # 
#88 142.4 # -- command failed --
#88 142.4 # status : 2
#88 142.4 # output (2 lines):
#88 142.4 #   # runtime
#88 142.4 #   /usr/lib/go-1.18/src/runtime/os_linux.go:206:40: undefined: syscall.SYS_FCNTL
#88 142.4 # --
#88 142.4 # 
#88 143.0 not ok 38 mips64le-softfloat-hellogo
#88 143.0 # (from function `assert_success' in file bats-assert/src/assert.bash, line 114,
#88 143.0 #  from function `testHelloGO' in file test-go.bats, line 240,
#88 143.0 #  in test file test-go.bats, line 329)
#88 143.0 #   `testHelloGO' failed with status 2
#88 143.0 # 
#88 143.0 # -- command failed --
#88 143.0 # status : 2
#88 143.0 # output (2 lines):
#88 143.0 #   # runtime
#88 143.0 #   /usr/lib/go-1.18/src/runtime/os_linux.go:206:40: undefined: syscall.SYS_FCNTL
#88 143.0 # --
#88 143.0 #
```

Ubuntu 22.04 ships Go 1.18.1: https://github.com/tonistiigi/xx/actions/runs/12236180951/job/34129079148?pr=176#step:5:1002

```
#83 89.73 # go version go1.18.1 linux/amd64
#83 89.73 ok 1 version
```

Issue looks similar to https://github.com/golang/go/issues/18031

--- a/src/test-go.bats
+++ b/src/test-go.bats
@@ -307,11 +307,17 @@ testHelloGO() {
 }
 
 @test "mipsle-hellogo" {
+  if ! supportMipsBuildGo; then
+    skip "MIPS build not supported"
+  fi
   export TARGETARCH=mipsle
   testHelloGO
 }
 
 @test "mipsle-softfloat-hellogo" {
+  if ! supportMipsBuildGo; then
+    skip "MIPS build not supported"
+  fi
   export TARGETARCH=mipsle
   export TARGETVARIANT=softfloat
   testHelloGO
@@ -319,11 +325,17 @@ testHelloGO() {
 }
 
 @test "mips64le-hellogo" {
+  if ! supportMipsBuildGo; then
+    skip "MIPS build not supported"
+  fi
   export TARGETARCH=mips64le
   testHelloGO
 }
 
 @test "mips64le-softfloat-hellogo" {
+  if ! supportMipsBuildGo; then
+    skip "MIPS build not supported"
+  fi
   export TARGETARCH=mips64le
   export TARGETVARIANT=softfloat
   testHelloGO

--- a/src/test_helper.bash
+++ b/src/test_helper.bash
@@ -95,6 +95,12 @@ supportAmd64VariantGo() {
   versionGTE "$(go version | awk '{print $3}' | sed 's/^go//')" "1.18"
 }
 
+# Supported since Go 1.13: https://go.dev/wiki/GoMips
+# But fails on Ubuntu 22.04 with Go 1.18.1: https://github.com/tonistiigi/xx/issues/177
+supportMipsBuildGo() {
+  ! grep -q 'ID=ubuntu' /etc/os-release && ! grep -q 'VERSION_ID="22.04"' /etc/os-release && versionGTE "$(go version | awk '{print $3}' | sed 's/^go//')" "1.13"
+}
+
 supportRC() {
   command -v llvm-rc >/dev/null 2>&1
 }

